Sources/ParseSwift/Parse Types/Internal/ParseLocationManager.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  ParseLocationManager.swift
+//  ParseSwift
+//
+//  Created by Corey Baker on 9/21/20.
+//  Copyright © 2020 Parse Community. All rights reserved.
+//
+
+import Foundation
+import CoreLocation
+#if canImport(UIKit)
+import UIKit
+#endif
+
+/**
+ ParseLocationManager is an internal class which wraps a CLLocationManager and
+ returns an updated CLLocation via the provided block.
+ When -addCurrentLocation is called, the CLLocationManager's
+ -startUpdatingLocations is called, and upon CLLocationManagerDelegate callback
+ (either success or failure), any handlers that were passed to this class will
+ be called _once_ with the updated location, then removed. The CLLocationManager
+ stopsUpdatingLocation upon a single failure or success case, so that the next
+ location request is guaranteed a speedily returned CLLocation.
+ */
+class ParseLocationMananger: NSObject {
```

Comment:
Is this functionality that we can / should be providing in this framework? It won't work on non-Apple platforms and it seems like a small enough thing that mostly everyone defines their own / uses Core Location directly.

Reply:
I ripped this and modified it from the objc SDK. That being said, it's not necessarily a requirement, as developers will just need to make an extra step in iOS to setup core location within their app and pass them to GeoPoint, which I'm okay with.

Do you think it should be removed?

Reply:
My main concern is that if we include it, we have to make sure it works, and it falls outside the core responsibility of this SDK. Plus, it's a pretty simple wrapper and we're not adding / extending Core Location in any big way.

Reply:
I agree, I will remove it